wasm/validation.cpp

```diff
@@ -0,0 +1,373 @@
+// SPDX-FileCopyrightText: 2024 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "wasm/validation.h"
+#include "util/variant.h"
+#include "wasm/instructions.h"
+#include "wasm/types.h"
+#include "wasm/wasm.h"
+
+#include <tl/expected.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <string_view>
+#include <variant>
+#include <vector>
+
+namespace wasm::validation {
+
+using namespace wasm;
```

Comment:
This doesn't do anything. When looking up symbols in `::wasm::validation`, the compiler will search in `::wasm::validation`, `::wasm`, `::`, in that order.